### PR TITLE
Added testing / js questions

### PR DESCRIPTION
**Pull request by hemstreet**
Singletons are useful when testing for dependency injection. This allows an object to be modified at run time.

Exporting an object as a singleton 

```
var MyFunction = function() {}
module.exports = new MyFunction();
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -159,6 +161,7 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What tools would you use to test your code's functionality?
 * What is the difference between a unit test and a functional/integration test?
 * What is the purpose of a code style linting tool?
+* What testing frameworks do you have experience with
```
Comment (appleboy) [inline README.md]: Missing `?`

File: README.md [hunk 2]
```diff
@@ -152,6 +152,8 @@ duplicate([1,2,3,4,5]); // [1,2,3,4,5,1,2,3,4,5]
 * What is event loop?
   * What is the difference between call stack and task queue?
 * Explain the differences on the usage of `foo` between `function foo() {}` and `var foo = function() {}` 
+* What is the benefits of singletons, how does this affect the unit tests can be written
+  * How would you export a function as a singleton
```
Comment (appleboy) [inline README.md]: Missing `?`

File: README.md (hunk 1 above)
Comment (hemstreet) [inline README.md]: Updated

File: README.md (hunk 2 above)
Comment (hemstreet) [inline README.md]: Updated

**Pull request closed without merging**